akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -472,6 +472,22 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
       "not exceed stream-level window while sending after SETTINGS_INITIAL_WINDOW_SIZE changed when window became negative through setting" in pending
 
       "eventually send WINDOW_UPDATE frames for received data" in pending
+
+      "reject WINDOW_UPDATE for connection with zero increment with PROTOCOL_ERROR" in new TestSetup with RequestResponseProbes {
+        sendWINDOW_UPDATE(0, 0) // illegal
+        val (_, errorCode) = expectGOAWAY()
+
+        errorCode should ===(ErrorCode.PROTOCOL_ERROR)
+      }
+      "reject WINDOW_UPDATE for stream with zero increment with PROTOCOL_ERROR" in new TestSetup with RequestResponseProbes {
+        // making sure we don't handle stream 0 and others differently here
+        sendHEADERS(1, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        sendWINDOW_UPDATE(1, 0) // illegal
+        val (_, errorCode) = expectGOAWAY()
```

Comment:
Done, here we get a RST_STREAM now, handled rather nicely IMO with new super type for exceptions on stream level, which carry streamid